zaas-client/src/main/java/org/zowe/apiml/zaasclient/service/internal/HttpsClientProvider.java

```diff
@@ -58,30 +63,30 @@ public HttpsClientProvider(ConfigProperties configProperties) throws ZaasConfigu
     }
 
     @Override
-    public CloseableHttpClient getHttpsClientWithTrustStore() throws ZaasConfigurationException {
-        return sharedHttpClientConfiguration(getSSLContext())
-            .build();
-    }
-
-    @Override
-    public CloseableHttpClient getHttpsClientWithTrustStore(BasicCookieStore cookieStore) throws ZaasConfigurationException {
-        return sharedHttpClientConfiguration(getSSLContext())
-            .setDefaultCookieStore(cookieStore)
-            .build();
+    public synchronized CloseableHttpClient getHttpsClientWithTrustStore() throws ZaasConfigurationException {
+        if (httpsClientWithTruststore == null) {
+            httpsClientWithTruststore = sharedHttpClientConfiguration(getSSLContext())
+                .build();
+        }
+        return httpsClientWithTruststore;
     }
 
+    //TODO how is this working compared to the other method to get the client?
```

Comment:
There is a TODO left. Beside that, it looks fine

Reply:
Yes, i left that TODO intentionally so it's not forgotten. I will address it with additional tests, but not in this PR. It does not have anything to do with performance. It's just the way the client is built which seems incorrect.

Reply:
If you agree, i'd keep it in and it will be removed later when i manage to verify if it's a problem. So far the code works, so i did not change it.

Reply:
I'm ok with that.